pkg/cmd/roachtest/tests/mixed_version_backup.go

```diff
@@ -2060,6 +2088,36 @@ func (d *BackupRestoreTestDriver) createBackupCollection(
 		}); err != nil {
 			return nil, err
 		}
+		backupEndTimes = append(backupEndTimes, latestIncBackupEndTime)
+
+		if d.testUtils.compactionEnabled && !collection.withRevisionHistory() && len(backupEndTimes) >= 3 {
+			// Require that endIdx - startIdx >= 2 so at least 2 inc backups are
+			// compacted. If there are 3 backupEndTimes, the start must be the the
+			// 0th. Thn endIdx is always the last index for now, so the compacted
+			// backup gets selected to restore.
```

Comment:
nit: For what it's worth, since we're doing a `RESTORE FROM LATEST` with no AOST, a compacted backup will always be selected as part of the restore chain.